core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -63,6 +67,28 @@ private void preparePartitionedTable() {
         .commit();
   }
 
+  @Test
+  public void testManifestsTableWithDroppedPartition() throws IOException {
```

Comment:
I think the key part of this test is we should be checking a dropped V1 Field whose return type changes when dropped. As is this test I believe would always pass on a V2 table even without this patch.

Reply:
It should be only in V1. I think it should be OK that the tests covered v2 as well, that's right?

Reply:
Yeah I just wanted to make sure we note that this test was added to check the transformation of a partition column's type. Like we are actually checking something a little more specific that dropping a partition, since if "data" was an int there wouldn't be a bug.